app/javascript/crayons/CTAs/CTA.jsx

```diff
@@ -0,0 +1,37 @@
+import { h } from 'preact';
+import classNames from 'classnames/bind';
+import { Icon } from '@crayons';
+
+export const CTA = (props) => {
+  const {
+    children,
+    href,
+    variant = 'default',
+    icon,
+    rounded,
+    className,
+    ...otherProps
+  } = props;
+
+
+  const classes = classNames('c-cta', {
+    [`c-cta--${variant}`]: variant,
+    'c-cta--icon-left': icon && children,
+    'c-cta--icon-alone': icon && !children,
+    'radius-full': rounded,
+    [className]: className,
+  });
+
+  return (
+    <a
+      href={href}
+      className={classes}
+      {...otherProps}
+    >
+      {icon && <Icon src={icon} className={classNames('c-cta__icon')} />}
```

Comment:
I think we are pretty much always going to want to add `aria-hidden="true"` to this icon element, as we can't rely on the `title` element being surfaced properly by screen readers.

Something we can consider from a design system POV is adding some guardrails around accessibility for things like icons - inspired by [this talk from Kathleen McMahon](https://www.youtube.com/watch?v=CgynK8rdsTY). 

Like, if a user passes an icon, but no other children and no aria-label, we could return `null` as a clear message to the developer that this thing isn't implementing our guidelines (which we would totally write 😆 ). We could also consider [a conditional "isRequired" prop types error](https://stackoverflow.com/questions/42299335/react-props-set-isrequired-on-a-prop-if-another-prop-is-null-empty).

Doesn't have to be covered in this PR, just raising as a discussion point, as it could form part of bringing these components out of 'beta', and could be a pretty impactful way to use our design system.

Reply:
yah, goood points!

Reply:
Yeah, agreed about the icon for `aria-hidden="true'`. I can't think of a scenario where it wouldn't be ornamental.